frontend/simple/js/backend/hapi.js

```diff
@@ -73,29 +73,55 @@ export class HapiBackend extends Backend {
   subscriptions () {
     return Object.keys(store.state.contracts)
   }
-  async subscribe (contractId: string) {
+  async subscribeAndSyncIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.subscribeAndSync(identity, ephemeral)
+  }
+  async unsubscribeIdentity (username: string, ephemeral: bool) {
+    let identity = await namespace.lookup(username)
+    await this.unsubscribe(identity, ephemeral)
+  }
```

Comment:
Specifically, this file only implements generic functions. If you start adding this kind of stuff, it becomes exponentially more difficult to switch backends.